analyses/tumor-purity-exploration/01_explore-tumor-purity.Rmd

```diff
@@ -0,0 +1,259 @@
+# broad_histology_display or cancer_group_display groups
+plot_tumor_violins <- function(df, display_name, display_hex) {
+  ggplot(df) + 
+    # order by ascending median
+    aes(x = forcats::fct_reorder({{display_name}}, tumor_fraction),
+        y = tumor_fraction, 
+        fill = {{display_hex}}) + 
+    geom_violin(alpha = 0.5) +
+    stat_summary(geom = "point", fun.y = "median") + 
+    scale_fill_identity() + 
+    theme(axis.text.x = element_text(angle = 30, hjust = 1)) 
+}
+```
+
+## Set directories and files
+
+```{r}
+# Input
+root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))
+data_dir <- file.path(root_dir, "data")
+pal_dir <- file.path(root_dir, "figures", "palettes")
+metadata_file <- file.path(data_dir, "pbta-histologies.tsv")
+pal_file <- file.path(pal_dir, "broad_histology_cancer_group_palette.tsv")
+
+# Output
+results_dir <- file.path("results")
+if (!(dir.exists(results_dir))) {
+  dir.create(results_dir)
+}
+rna_stranded_df_file <- file.path(results_dir, "rna_stranded_same-extraction.tsv")
+```
+
+
+Read in:
+
+```{r}
+metadata <- read_tsv(metadata_file) 
+palette_mapping_df <- read_tsv(pal_file)
+```
+
+## Exploration of `tumor_fraction`
+
+The quantity `tumor_fraction` provides information about what percent of the sample was likely tumor. 
+We'll explore this variable:
+
+```{r}
+# Note that this quantity is only recorded, if ever, with WGS libraries: 
+presence_table <- table(metadata$experimental_strategy,
+                        is.na(metadata$tumor_fraction))
+colnames(presence_table) <- c("Tumor fraction known", "Tumor fraction unknown")
+presence_table
+```
+
+Tumor fraction was calculated for WGS samples:
+
+```{r}
+metadata_tumor_wgs <- metadata %>%
+  filter(experimental_strategy == "WGS", 
+         sample_type == "Tumor")
+
+# For what percentage of tumor samples is purity known? Nearly all of them!
+sum(!is.na(metadata_tumor_wgs$tumor_fraction)) / nrow(metadata_tumor_wgs)
+```
+
+
+Distribution of tumor purity:
+
+```{r}
+summary(metadata_tumor_wgs$tumor_fraction)
+
+ggplot(metadata_tumor_wgs) + 
+  aes(x = tumor_fraction) + 
+  geom_histogram(color = "black", fill = "skyblue")
+```
+
+Distribution by cancer group and broad histology, where the point in each plot is the median value (and groups are ordered by median).
+
+
+```{r}
+# combine metadata with palette:
+tumor_palette_df <- metadata_tumor_wgs %>%
+  inner_join(palette_mapping_df) %>%
+  select(Kids_First_Biospecimen_ID, 
+         tumor_fraction, 
+         cancer_group_display, 
+         cancer_group_hex, 
+         broad_histology_display,
+         broad_histology_order,
+         broad_histology_hex) %>%
+  drop_na() %>%
+  # reorder by median fraction and ensure "Other" is always last
+  mutate(cancer_group_display = forcats::fct_reorder(cancer_group_display, tumor_fraction), 
+         cancer_group_display = forcats::fct_relevel(cancer_group_display, "Other", after = Inf),
```

Comment:
Making sure Other is last is about to get "undone" by the function `plot_tumor_violins()`. I'd either remove this and the analogous broad histology step or move putting Other last into the function. I actually think what we might want is the option to drop Other that's `TRUE` by default in `plot_tumor_violins()` – the category is broad enough as to be confusing in this plot – and order by median no matter what. Alternatively, drop Other before you `plot_tumor_violins()` (this might be what I favor given the ease of implementation).